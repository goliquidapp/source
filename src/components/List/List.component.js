import React, {Component} from 'react';
import { View, ScrollView, Text, TouchableOpacity} from 'react-native';
import {Icon} from 'react-native-elements';

import Theme from '../../resources/Theme.js';

class List extends Component{

	render(){
		const {style, visible,onClose, options, selected, onChange}=this.props;
		if (!visible) return <View></View>
		return (
			<View style={[styles.container,style]}>
				<TouchableOpacity style={styles.close} onPress={onClose}>
					<Icon name={'close'} color={Theme['dark'].secondaryText}/>
				</TouchableOpacity>
				<ScrollView nestedScrollEnabled={true}>
					{options.map((option,index)=>{
							if (selected.indexOf(option)>=0){
								return (
										<TouchableOpacity key={index.toString()} style={styles.selectedItem} onPress={()=>onChange(option,false)}>
											<Text style={styles.textItemSelected}>{option}</Text>
										</TouchableOpacity>
									)
							}
							else{
								return (
										<TouchableOpacity key={index.toString()} style={styles.item} onPress={()=>onChange(option,true)}>
											<Text style={styles.textItem}>{option}</Text>
										</TouchableOpacity>
									)
							}
						})}
				</ScrollView>	
			</View>
		)
	}
}

const styles={
	container:{
		justifyContent:'center',
		width:'80%',
		height:400,
		position:'absolute',
		top:'20%',
		left:'10%',
		elevation:5,
		zIndex:11,
		backgroundColor:Theme['dark'].disabled
	},
	item:{
		zIndex:12,
		paddingVertical:5,
		borderBottomColor:Theme['dark'].border,
		borderBottomWidth:1,
		backgroundColor:Theme['dark'].disabled
	},
	selectedItem:{
		zIndex:12,
		paddingVertical:5,
		borderBottomColor:Theme['dark'].border,
		borderBottomWidth:1,
		backgroundColor:Theme['dark'].highlighted
	},
	textItem:{
		fontSize:14,
		padding:20,
		color:Theme['dark'].primaryText
	},
	textItemSelected:{
		fontSize:14,
		padding:20,
		color:Theme['dark'].primaryText
	},
	close:{
		top:5,
		right:5,
		position:'absolute',
		zIndex:13,
		padding:10
	}
}

export default List