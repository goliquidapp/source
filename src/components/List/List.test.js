import React from 'react';
import { Text, View, TouchableOpacity, ScrollView} from 'react-native';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import List from './List.component';

describe("Testing List component",()=>{
	it('renders correctly', () => {
		const options=[
			'option1',
			'option2',
			'option3'
		]
		const wrapper = renderer.create(
    		<List visible={true} options={options} selected={'option2'}/>
    	).toJSON();

		expect(wrapper).toMatchSnapshot();
	});

	it('opens when changing visible prop', () => {
		const options=[
			'option1',
			'option2',
			'option3'
		]
		const wrapper = shallow(
    		<List visible={false} options={options} selected={'option2'}/>
    	);

		expect(wrapper.find(ScrollView).length).toEqual(0);

		wrapper.setProps({ visible: true });

		expect(wrapper.find(ScrollView).length).toEqual(1);
	});
})