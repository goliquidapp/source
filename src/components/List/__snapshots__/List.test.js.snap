// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "elevation": 5,
        "height": 400,
        "justifyContent": "center",
        "left": "10%",
        "position": "absolute",
        "top": "20%",
        "width": "80%",
        "zIndex": 11,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 10,
        "position": "absolute",
        "right": 5,
        "top": 5,
        "zIndex": 13,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
        underlayColor="white"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#96969C",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          Óóç
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    nestedScrollEnabled={true}
  >
    <View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "#6D6C6C",
            "borderBottomWidth": 1,
            "opacity": 1,
            "paddingVertical": 5,
            "zIndex": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "padding": 20,
            }
          }
        >
          option1
        </Text>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5367FF",
            "borderBottomColor": "#6D6C6C",
            "borderBottomWidth": 1,
            "opacity": 1,
            "paddingVertical": 5,
            "zIndex": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "padding": 20,
            }
          }
        >
          option2
        </Text>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "#6D6C6C",
            "borderBottomWidth": 1,
            "opacity": 1,
            "paddingVertical": 5,
            "zIndex": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "padding": 20,
            }
          }
        >
          option3
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
